### ISSUE TITLE
InstallFullDB.sh doesn't work on a Ubuntu

### ISSUE THREAD

**Issue by adellaci** [unified-db/Database#158]
I was unable to get InstallFullDB.sh to run on a Ubuntu 14.04 system. I have not tested it on any other Linux systems. The changes I have made to the script now allows it to run.
[InstallFullDB_Linux.txt](https://github.com/unified-db/Database/files/72685/InstallFullDB_Linux.txt)

The *.sql files process correctly. There are comments next to all the lines changed

**Comment by Grz3s** [comment]
do we need to see this??? ( personlay i like when is short and clear)....
```code
## Full
echo "Process Full_DB ${ADDITIONAL_PATH}Current_Release/Full_DB/UDB_0.12.2_mangos_11792_SD2_2279.sql"      # Show exact location and file being processed
$MYSQL_MANGOSDB_CMD < ${ADDITIONAL_PATH}Current_Release/Full_DB/UDB_0.12.2_mangos_11792_SD2_2279.sql
echo "Test1"
# [[ $? != 0 ]] && exit 1                                                                                  # Commented out
echo $?
```

??

**Comment by Grz3s** [comment]
pls look at this one... and let me know if:
- works fine on linux
- its ok for you.

http://paste2.org/Zdc56dEI

**Comment by adellaci** [comment]
I will be back at the house at 5:00 pm EST I'll ran and respond right away


Sent from my Verizon Wireless 4G LTE smartphone

-------- Original message --------

**Comment by adellaci** [comment]
Grz3s,
Yes looks a lot cleaner. I also cleaned up some other comments.
I ran it after all the changes and it appears every ran correctly.
I pushed it up again.
https://github.com/adellaci/Database

I left it named as InstallFullDB_Linux.sh, in case the original works for other OS's.

I guess I should have explain, in comment, why I changed it.
the statements 'function create_config {' & 'Function display_help {'
generated errors when run on LInux of;
./InstallFullDB.sh: 23: ./InstallFullDB.sh: function: not found
./InstallFullDB.sh: 65: ./InstallFullDB.sh: Syntax error: "}" unexpected

Once that was fixed it generated an error;
InstallFullDB.sh: 85: .: InstallFullDB.config: not found

That is when I changed these statements to this;
SCRIPT_FILE="./InstallFullDB.sh"
CONFIG_FILE="./InstallFullDB.config"

Once that was fixed,  every statements   '[[ $? != 0 ]] && exit 1'
generated an output of; 
./InstallFullDB.sh: 107: ./InstallFullDB.sh: [[: not found
./InstallFullDB.sh: 112: ./InstallFullDB.sh: [[: not found
.......and so on

I believe that was for the progress bar.

**Comment by Grz3s** [comment]
thanks.. for explain...
Let's keep only one file.
well im using win7... and this ver works for me also well.
So... we should be fine with this.

I'll leave this for few days.. and let other ppl (with diff systems) look at this and report if anything is wrong.
If noone will complain about it ..ill push it ;)
thanks

**Comment by adellaci** [comment]
That sound good, thank you., and happy new year if i don't talk to you by then.

**Comment by Schmoozerd** [comment]
Once that was fixed, every statements '[[ $? != 0 ]] && exit 1'
generated an output of;
./InstallFullDB.sh: 107: ./InstallFullDB.sh: [[: not found
./InstallFullDB.sh: 112: ./InstallFullDB.sh: [[: not found
.......and so on, thou it processed all the sql files correctly at this point.

I believe that was for the progress bar. 

The [[ &? != 0 ]] && exit statement is a (simple) statement to quit on errors.
In my opinion this should be included (especially as in case of errors the first error is the important one)

**Comment by Grz3s** [comment]
thx @adellaci.
I'll wait for your feedback then. 

Hi @Schmoozerd, nice to see you again here..
if You know how make Win, Linux and other systems happy with... this code... I'll add anything you like. (you are an expert with this) ;)

**Comment by adellaci** [comment]
Hi Grz3s & Schmoozerd

I agree that itshould be added back, but as it was, it would generate the warning as i showed. Even thou the warning was showing it did process the sql statements correctly.  Off the top of your head, would know how we could alter the statement to that it doesn't create the warning, yet still function as intended? In the mean while i will try to researrch it.

**Comment by adellaci** [comment]
This is just a quick option, but would this work? I'm not at home to try it.

if [  $? -eq 1 ]
then 
    echo "File failed to process"
    exit 1
fi

or:

if [ $? -eq 0 ]
then
    echo " File processed successfully"
else
    echo " File failed to process" 
    exit 1
fi


I don't know if it is possible, instead of 'exit 1' could 'exit $?' be used?

**Comment by adellaci** [comment]
I did n't get it till now. Without the 'if", is why  it generated the responce it did. I'll test it as soon as i get home tomorrow.

**Comment by Schmoozerd** [comment]
This statement uses sloppy evaluation with the AND (&&) operator
if the first statement evaluates to TRUE, the second statement is evaluated (and thereby executed)
So yes, it is equivalent to
if [[ $? != 0 ]]; then exit 1;

The amount of parantheses [[ vs. [ - there i am not sure what the expected standard is (I know we hit this in the past a few times..) Git bash was happy with both versions, but not all versions of *sh* on *nix systems were.
also the #!/bin/sh vs #!/bin/bash in the first line makes sometimes a difference here :(

--

Within the "exit 1" statement the 1 is only used as marker to return something else then 0, as 0 usually is evaluated as "without errors". Of course you could also pass the sql-error code with $? as output, but i doubt there is much use :)

--

hi @Grz3s  :)

**Comment by adellaci** [comment]
Hi Schmoozerd & Grz3s
Thank you both for your input.

I ran a test with the statement as "if [[ $? != 0 ]]; then exit 1;" /w "#!/bin/sh". It complained about "[[", so reduced it to "if [ $? != 0 ]; then exit 1;" and it complained about no "fi". I then changed it to;
if [ $? != 0 ]
	then exit 1
fi
and that worked.(still haven't created an error in the sql files at this point.)

then I changed it to;
if [ $? != 0 ]; then exit 1; fi;
and it like this also

I then created and error (PROP instead of DROP) in 'UDB_0.12.2_mangos_11792_SD2_2279.sql' file and it exited with;
ERROR 1064 (42000) at line 23: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'PROP TABLE IF EXISTS `achievement_criteria_requirement`' at line 1 

I  then changed the line in 'InstallFullDB.sh' to '#!/bin/bash', and ran it as 'bash ./InstallFullDB.sh', and it ran successfully, both with the sql file correct and with an error. Both ran as expected.

The only difference I found, was when I ran it as 'bash ./InstallFullDB.sh', I actually got a count of "."s for the 20 seconds to hit Ctrl+C.

SUGGESTION; 
We use the statement 'if [ $? != 0 ]; then exit 1; fi;',  '#!/bin/bash', and execute it as 'bash ./InstallFullDB.sh'. That seems to run the closest to what was intended.

Ya or Na?
